src/sql/workbench/contrib/notebook/browser/notebookViews/insertCellsModal.ts

```diff
@@ -0,0 +1,326 @@
+		if (this._optionsMap[optionName]) {
+			this._optionsMap[optionName].currentValue = value;
+		}
+	}
+
+	public getOptionValue(optionName: string): any {
+		if (this._optionsMap[optionName]) {
+			return this._optionsMap[optionName].currentValue;
+		}
+		return undefined;
+	}
+}
+
+export class InsertCellsModal extends Modal {
+	public viewModel: CellOptionsModel;
+
+	private _submitButton: Button;
+	private _cancelButton: Button;
+	private _optionsMap: { [name: string]: Checkbox } = {};
+
+	constructor(
+		private onInsert: (cell: ICellModel) => any,
+		private _context: NotebookViewsExtension,
+		private _containerRef: ViewContainerRef,
+		private _componentFactoryResolver: ComponentFactoryResolver,
+		@ILogService logService: ILogService,
+		@IThemeService themeService: IThemeService,
+		@ILayoutService layoutService: ILayoutService,
+		@IClipboardService clipboardService: IClipboardService,
+		@IContextKeyService contextKeyService: IContextKeyService,
+		@IAdsTelemetryService telemetryService: IAdsTelemetryService,
+		@ITextResourcePropertiesService textResourcePropertiesService: ITextResourcePropertiesService,
+	) {
+		super(
+			localize("insertCellsModal.title", "Insert cells"),
+			'InsertCellsModal',
+			telemetryService,
+			layoutService,
+			clipboardService,
+			themeService,
+			logService,
+			textResourcePropertiesService,
+			contextKeyService,
+			{ hasErrors: true, hasSpinner: true }
+		);
+
+		const options = this.getOptions();
+		this.viewModel = new CellOptionsModel(options, this.onInsert, this._context);
+	}
+
+	protected renderBody(container: HTMLElement): void {
+		const grid = DOM.$<HTMLDivElement>('div#insert-dialog-cell-grid');
+
+		grid.style.display = 'grid';
+		grid.style.gridTemplateColumns = '1fr 1fr';
+		grid.style.gap = '10px';
+		grid.style.padding = '10px';
+		grid.style.overflowY = 'auto';
+		grid.style.maxHeight = 'calc(100% - 40px)';
+
+		const gridTitle = DOM.$<HTMLHeadElement>('h2.grid-title');
+		gridTitle.title = localize("insertCellsModal.selectCells", "Select cell sources");
+
+		DOM.append(container, grid);
+
+		this.createOptions(grid)
+			.then(() => { })
+			.catch((e) => { this.setError(localize("insertCellsModal.thumbnailError", "Error: Unable to generate thumbnails.")); });
+	}
+
+	protected layout(height: number): void {
+		// No-op for now. No need to relayout.
+	}
+
+	private async createOptions(container: HTMLElement): Promise<void> {
+		const activeView = this._context.getActiveView();
+		const cellsAvailableToInsert = activeView.hiddenCells;
+
+		this._themeService.getColorTheme().getColor(editorBackground);
+
+		cellsAvailableToInsert.forEach(async (cell) => {
+			const optionWidget = this.createCheckBoxHelper(
+				container,
+				'<div class="loading-spinner-container"><div class="loading-spinner codicon in-progress"></div></div>',//cell.renderedOutputTextContent[0]?.substr(0, 20) ?? localize("insertCellsModal.untitled", "Untitled Cell : {0}", cell.cellGuid),
```

Comment:
remove commented out code

Reply:
Why are you passing HTML content as the "label"?

Reply:
This is related to https://github.com/microsoft/azuredatastudio/pull/16043, where I was trying to allow the checkbox to allow html so that I could add an image thumbnail as a label. I left this here while I work on the new approach which you have described here https://github.com/microsoft/azuredatastudio/pull/16467. 

I should be able to plug the new solution here. Leaving this here causes no problems, it just shows the html as text in the dialog for Notebook Views, which is currently not even accessible.